commit abf3fa4b0b3b4201ef27c7c608f2fd86e4193b57 (refs/changes/21/3721/2)
Author: Josh Wilsdon <josh@wilsdon.ca>
Date:   2018-03-24T12:42:46-07:00 (1 year, 7 months ago)
    
    OS-6825 should preserve zhyve.log when archiving log files from deleted bhyve VMs
    Reviewed by: Todd Whiteman <todd.whiteman@joyent.com>
    Approved by: Todd Whiteman <todd.whiteman@joyent.com>

diff --git a/src/vm/node_modules/VM.js b/src/vm/node_modules/VM.js
index f6679e14..09a53ef3 100644
--- a/src/vm/node_modules/VM.js
+++ b/src/vm/node_modules/VM.js
@@ -6517,6 +6517,11 @@ function archiveVM(uuid, options, callback)
                     targ: path.join(archive_dirname, 'startvm')
                 });
             } else if (vmobj.brand === 'bhyve') {
+                patterns_to_archive.push({
+                    src: path.join(vmobj.zonepath, 'root/tmp/zhyve.log'),
+                    dst: path.join(archive_dirname, 'vmlogs'),
+                    create_dst_dir: true
+                });
                 patterns_to_archive.push({
                     src: path.join(vmobj.zonepath, 'root/var/run/*.log*'),
                     dst: path.join(archive_dirname, 'vmlogs'),
